src/models/tree/build_tree.cc

```diff
@@ -298,7 +298,12 @@ struct TreeBuilder {
       }
     }
   }
-  void PerformBestSplit(Tree& tree, Histogram<GPair> histogram, double alpha, NodeBatch batch)
+  void PerformBestSplit(Tree& tree,
+                        Histogram<GPair> histogram,
+                        double l2_regularization,
+                        NodeBatch batch,
+                        double l1_regularization,
```

Comment:
```suggestion
                        NodeBatch batch,
                        double l2_regularization,
                        double l1_regularization,
```
maybe always l2 + l1 together (in that order)

Reply:
I'll switch it to l1, l2.